Microsoft.Azure.CosmosRepository/src/DefaultRepository.cs

```diff
@@ -279,6 +284,60 @@ public async ValueTask<bool> ExistsAsync(Expression<Func<TItem, bool>> predicate
             return count > 0;
         }
 
+        public async ValueTask<IPage<TItem>> ScrollAsync(
+            Expression<Func<TItem, bool>> predicate = null,
+            int pageSize = 25,
+            string continuationToken = null,
+            CancellationToken cancellationToken = default)
+        {
+            Container container = await _containerProvider.GetContainerAsync().ConfigureAwait(false);
+
+            QueryRequestOptions options = new()
+            {
+                MaxItemCount = pageSize
+            };
+
+            IQueryable<TItem> query = container
+                .GetItemLinqQueryable<TItem>(requestOptions: options, continuationToken: continuationToken)
+                .Where(_repositoryExpressionProvider.Build(predicate));
+
+            Response<int> countResponse = await query.CountAsync(cancellationToken);
+
+            List<TItem> results = new();
+
+            int readItemsCount = 0;
+
+            using FeedIterator<TItem> iterator = query.ToFeedIterator();
+
+            double charge = countResponse.RequestCharge;
+
+            while (readItemsCount < pageSize && iterator.HasMoreResults)

```

Comment:
```suggestion
            Response<int> countResponse = await query.CountAsync(cancellationToken);
            using FeedIterator<TItem> iterator = query.ToFeedIterator();

            List<TItem> results = new();
            int readItemsCount = 0;
            double charge = countResponse.RequestCharge;

            while (readItemsCount < pageSize && iterator.HasMoreResults)
```

I think that some of the extra whitespace and new lines made it harder to read. I grouped a few things together and slightly re-orged it a bit.

Reply:
Looks good to me